tests/integration/targets/user/tasks/main.yml

```diff
@@ -0,0 +1,115 @@
+---
+- environment:
+    SC_HOST: "{{ sc_host }}"
+    SC_USERNAME: "{{ sc_username }}"
+    SC_PASSWORD: "{{ sc_password }}"
+    SC_TIMEOUT: "{{ sc_timeout }}"
+
+  block:
+    - name: Delete user - cleanup
+      scale_computing.hypercore.user:
+        state: absent
+        username: "{{ item }}"
+      loop:
+        - my_user
+        - my_user_updated 
+
+    - name: Create user
+      scale_computing.hypercore.user:
+        state: present
+        username: my_user
+        password: my_password
+        full_name: my_full_name
+        roles:
+          - Admin
+        session_limit: 2
+      register: user
+    - ansible.builtin.assert:
+        that:
+          - user is changed
+          - user.record.keys() | sort == ['full_name', 'roles', 'session_limit', 'username', 'uuid']
+          - user.record.username == "my_user"
+          - user.record.full_name == "my_full_name"
+          - user.record.roles[0]["name"] == "Admin"
+          - user.record.session_limit == 2
```

Comment:
You should use also `user_info` module to check what is actual stat on server. Checking content of `user` variable only checks what user module thinks was setup on the HC3.

I need to look into python code to see if you read new state from server or not. And if remove the line that does it - the test would not detect it.

Reply:
Thanks for the comment, I agree with you, I added checks with user_info module